ultradns-rest/src/main/java/denominator/ultradns/service/UltraDNSRestResourceRecordSetApi.java

```diff
@@ -0,0 +1,311 @@
+  }
+
+  /**
+   * Updates resource record(s) given rrSetList, resource record set, type, ttlToApply.
+   *
+   * @param rrSetList
+   * @param rrset
+   * @param ttlToApply
+   * @param typeCode
+   * @throws IllegalArgumentException if the zone is not found.
+   */
+  private void replace(RRSetList rrSetList, ResourceRecordSet<?> rrset , int ttlToApply, int typeCode) {
+    LOGGER.debug("Updating resource record(s) of the zone:" + zoneName + " domain name:" + rrset.name()
+            + " type:" + typeCode);
+    // Update the resource record set.
+    List<RRSet> rrSetsExisting = rrSetList.rrSets();
+    RRSet rrSetExisting = rrSetsExisting.remove(0);
+    // Set the TTL value in the resource record set.
+    rrSetExisting.setTtl(ttlToApply);
+    setData(rrSetExisting, rrset);
+    // Send a PUT request.
+    api.updateResourceRecord(zoneName, typeCode, rrset.name(), rrSetExisting);
+  }
+
+  /**
+   * Updates the passed rdatas to the existing RRSet.
+   *
+   * @param rrSet RRSet
+   * @param rrset ResourceRecordSet
+   * @throws IllegalArgumentException if the zone is not found.
+   */
+  private void setData(RRSet rrSet, ResourceRecordSet<?> rrset) {
+    List<String> rdatas = new ArrayList<String>();
+    // Extract the rdata from the resource record set passed in as
+    // the parameter to this method and put them into rdatas.
+    List<Map<String, Object>> toCreate = new ArrayList<Map<String, Object>>(rrset.records());
+    for (Iterator<Map<String, Object>> shouldCreate = toCreate.iterator(); shouldCreate.hasNext();) {
+      String rdata = flatten(shouldCreate.next());
+      rdatas.add(rdata);
+    }
+    rrSet.setRdata(rdatas);
+  }
+
+  /**
+   * Returns RRSetList given owner name and type.
+   *
+   * @param dName
+   * @param type
+   * @throws IllegalArgumentException if the zone is not found.
+   */
+  private RRSetList getRrSetList(String dName, int type) {
+    RRSetList rrSetList;
+    try {
+      rrSetList = api.getResourceRecordsOfDNameByType(zoneName, dName, type);
+    } catch (UltraDNSRestException e) {
+      if (e.code() == UltraDNSRestException.DATA_NOT_FOUND ||
+              e.code() == UltraDNSRestException.RESOURCE_RECORD_POOL_NOT_FOUND) {
+        rrSetList = null;
+      } else {
+        throw e;
+      }
+    }
+    return rrSetList;
+  }
+
+  /**
+   * Deletes records for the specified name and type.
+   *
+   * @param name
+   * @param type
+   */
+  @Override
+  public void deleteByNameAndType(String name, String type) {
+    LOGGER.debug("Deleting resource record(s) for the zone:" + zoneName + " domain name:" + name + " type:" + type);
+    for (Record record : recordsByNameAndType(name, type)) {
+      remove(name, type, record);
+    }
+  }
+
+  /**
+   * Deletes record for the specified name and type.
+   *
+   * @param name
+   * @param type
+   * @param record
+   */
+  private void remove(String name, String type, Record record) {
+    int intType = lookup(type);
```

Comment:
No, you don;t need all this complex logic. Just the logic to call deleteResourceRecord is sufficient. You don't need to call GET before this. Also, you don't need any specific handling to check if it is a pool. In terms of DELETE REST API, deleting a pool is same as deleting a record. The URI is exactly the same. Also, make sure the logic present under UltraDNSRestRoundRobinPoolApi#deletePool is not duplicated. Also, remove the method UltraDNSRest#deleteLBPool, and use UltraDNSRest#deleteResourceRecord instead.